### PR TITLE
Fix incorrect preview URL for cropped images

### PR DESCRIPTION
**Pull request by anton-vlasenko**
Trac ticket: https://core.trac.wordpress.org/ticket/55070

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by anton-vlasenko** [comment]
Steps to test this PR: https://core.trac.wordpress.org/ticket/55070#comment:6

File: src/wp-includes/media-template.php [hunk 1]
```diff
@@ -357,8 +357,23 @@ function wp_print_media_templates() {
 			<div class="thumbnail thumbnail-{{ data.type }}">
 				<# if ( data.uploading ) { #>
 					<div class="media-progress-bar"><div></div></div>
-				<# } else if ( data.sizes && data.sizes.large ) { #>
-					<img class="details-image" src="{{ data.sizes.large.url }}" draggable="false" alt="" />
+				<# } else if ( data.sizes && data.sizes.large ) {
+					var large_image_preview_url = data.sizes.large.url;
+					if ( data.sizes.full ) {
+						var full_image_crop_id = data.sizes.full.url.match(/^.+-(e[0-9]{13})\.[^\\/]+$/)?.[1],
```
Comment (anton-vlasenko) [inline src/wp-includes/media-template.php:363]: I hope it's OK to use optional chaining operator here (`?.`).
According to [this page](https://make.wordpress.org/core/handbook/best-practices/browser-support/) all browsers supported by WordPress should support this syntax.

File: src/wp-includes/media-template.php [hunk 2]
```diff
@@ -357,8 +357,23 @@ function wp_print_media_templates() {
...
+							large_image_crop_id = data.sizes.large.url.match(/^.+-(e[0-9]{13})-[^\\/]+$/)?.[1];
```
Comment (anton-vlasenko) [inline src/wp-includes/media-template.php:364]: Same. See my [comment](https://github.com/WordPress/wordpress-develop/pull/2489/files#r840024804) above.

File: src/wp-includes/media-template.php [hunk 3]
```diff
@@ -357,8 +357,23 @@ function wp_print_media_templates() {
...
+						<?php
+							// A cropped image should be used if the "large" image hasn't been updated during cropping.
+							// It can happen if the original image is larger than the cropped image and
+							// the cropped image is smaller than 1024x1024 pixels (the size of a "large" image).
```
Comment (ironprogrammer) [inline src/wp-includes/media-template.php]: I suggest removing the default pixel dimensions (1024x1024) and only refer to the "Large size" as configured in *Settings > Media Settings*. (Through testing I confirmed that images smaller than the configured large dimensions were subject to the reported issue, so the exact pixel sizes may vary.)

Comment (anton-vlasenko) [reply src/wp-includes/media-template.php]: Thank you for the review, @ironprogrammer.
That's good point. Fixed in aa66d7cd22ae523fbe790ac720e6d0b70bb7577d

File: src/wp-includes/media-template.php (hunk 1 above)
Comment (adamsilverstein) [reply src/wp-includes/media-template.php:363]: @anton-vlasenko this regex is run in PHP, right? so the PHP engine is what needs to support the optional chaining. which, I think it does for versions we support. Adding some test cases that get run across PHP versions would help ensure that

Comment (anton-vlasenko) [reply src/wp-includes/media-template.php:363]: Thank you for the review, @adamsilverstein.

> this regex is run in PHP, right? 

This regex is [run in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match).
According to [this page (please open the `Date relative` tab)](https://caniuse.com/mdn-javascript_operators_optional_chaining), **the vast majority** of the browsers released in 2020 and later support this syntax.

Comment (adamsilverstein) [reply src/wp-includes/media-template.php:363]: that covers all our supported browsers so should be fine. what are the consequences if it fails? console error?

Comment (anton-vlasenko) [reply src/wp-includes/media-template.php:363]: @adamsilverstein Yes, console error.
I've just checked the source code, and the `?.` operator is being used in the `src/wp-includes/js/dist/components.js`.
So I would assume it's safe to use it.

**Comment by anton-vlasenko** [comment]
Closed in favor of https://github.com/WordPress/wordpress-develop/pull/3985.

**Pull request closed without merging**